### PR TITLE
added custom sort_by, order and filter for topics

### PR DESCRIPTION
**Pull request by adampaulsackfield**

### CODE REVIEW

File: __tests__/article.test.js [hunk 1]
```diff
@@ -118,36 +146,45 @@ describe('ARTICLES', () => {
...
 				});
 		});
 
+		it('should be able to filter topics', () => {
+			return request(app)
+				.get(`${ENDPOINT}?topic=cats`)
+				.expect(200)
+				.then((res) => {
+					expect(res.body.articles).toBeInstanceOf(Array);
+					expect(res.body.articles[0].topic).toBe('cats');
```
Comment (jadelandeg) [inline __tests__/article.test.js:186]: check the length of the array is correct

Comment (jadelandeg) [reply __tests__/article.test.js:186]: nice tests. Good start! Now think about what could go wrong with the queries and handle those errors. Invalid sortby etc....
Also what happens if a topic doesn't exist?

File: models/article.model.js [hunk 2]
```diff
@@ -1,11 +1,41 @@
...
+	}
+
+	query.text += `GROUP BY articles.article_id ORDER BY ${sort_by} ${order};`;
+	return db.query(query).then((result) => {
+		if (!result.rows.length) {
+			return Promise.reject({ status: 400, msg: 'Articles not found' });
+		}
+		return result.rows;
+	});
 };
```
Comment (jadelandeg) [inline models/article.model.js:39]: building up your query really nicely here, good job! I think you might be missing some validation in your order by leaving you vulnerable for SQL injection there

**Review by jadelandeg — CHANGES_REQUESTED** [review]
well done! Getting on really well, just a couple of additions to be made

**Review by karlaferry — APPROVED** [review]
Very good! No more additional comments from me. Feel free to merge!

**Pull request merged**